src/Sylius/Bundle/WebBundle/Resources/views/Backend/Product/show.html.twig

```diff
@@ -234,6 +234,35 @@
                 {% endfor %}
             </tbody>
         </table>
+        <table class="table table-bordered">
+            <thead>
+            <tr>
+                <th colspan="2">{{ 'sylius.product.associations'|trans }}</th>
+            </tr>
+            </thead>
+            <tbody>
+            {% for association in product.associations %}
+                <tr>
+                    <td class="sylius-association-type"><strong>{{ association.type.name }}</strong></td>
+                    <td>
+                        <ul>
+                        {% for associatedObject in association.associatedObjects %}
```

Comment:
If we assume that associated object has name, we can assume that it is product as well: `{% for associatedProduct in association.associatedObjects %}`